src/main/kotlin/org/rust/ide/actions/ToggleCargoCheckAction.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import org.rust.cargo.project.settings.rustSettings
+
+class ToggleCargoCheckAction : AnAction() {
+
+    companion object {
+        private const val disableText = "Disable Cargo Check"
+        private const val enableText = "Enable Cargo Check"
+    }
+
+    override fun update(e: AnActionEvent) {
+        val project = e.project
+        if (project == null) {
+            e.presentation.isEnabledAndVisible = false
+            return
+        }
+
+        val isHintsShownNow = project.rustSettings.useCargoCheckAnnotator
+        e.presentation.text = if (isHintsShownNow) disableText else enableText
```

Comment:
Are we sure about dynamic presentation text?
At this moment it's rather hard to find the action via `find action` action because it can be found by "Disable Cargo Check" or "Enable Cargo Check". At the same time when I type "Toggle Cargo check" (or part of it) in the corresponding dialog, I expect to find the same text in suggested items but because of different dynamic presentation text I fail to do it
![image](https://user-images.githubusercontent.com/2539310/52673736-a7b46b00-2f32-11e9-8de8-464875a814af.png)
Maybe it's better add toggle button as some other actions do?